lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -68,7 +81,7 @@ internal sealed class MapTaskHost<TMapInput, TMapOutput> : ITask
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            while (true)
+            while (Interlocked.Read(ref _toClose) == 0)
```

Comment:
My understanding is Interlocked only lock that variable when accessing it. Once entering the while loop, the lock should have been released.